packages/@lwc/synthetic-shadow/src/faux-shadow/slot.ts

```diff
@@ -82,6 +93,80 @@ function getFilteredSlotFlattenNodes(slot: HTMLElement): Node[] {
     );
 }
 
+interface SlotMeta {
```

Comment:
on every slot element, we track the current state of the distribution, while keeping the default content in a fragment when the slotted content is encounter in case we need to bring it back.